src/lamplib/src/genny/tasks/run_tests.py

```diff
@@ -203,78 +210,41 @@ def _setup_resmoke(
-            "Found existing mongod so will not download artifacts.",
-            existing_mongod=mongod,
-            wont_download_artifacts_from=mongodb_archive_url,
-        )
-
-    if mongod is None:
-        SLOG.info(
-            "Couldn't find pre-build monogod. Fetching and installing.",
-            looked_at=(opt, install, from_tarball),
-            fetching=mongodb_archive_url,
-        )
-        if mongodb_archive_url is None:
-            info = toolchain.toolchain_info(
-                genny_repo_root=genny_repo_root, workspace_root=workspace_root
-            )
-
-            if info.is_darwin:
-                artifact_key = "osx"
-            elif info.linux_distro == "amazon2":
-                artifact_key = "amazon2"
-            else:
-                raise Exception(
-                    f"No pre-built artifacts for distro {info.linux_distro}. You can either:"
-                    f"1. compile/install a local mongo checkout in ./src/mongo."
-                    f"2. Modify the _canned_artifacts dict in the genny python to include an artifact from a waterfall build."
-                    f"3. Pass in the --mongodb-archive-url parameter to force a canned artifact."
-                )
-            mongodb_archive_url = _canned_artifacts[artifact_key]
-
-            cmd_runner.run_command(
-                cmd=["curl", "-LSs", mongodb_archive_url, "-o", "mongodb.tgz"],
-                cwd=mongo_repo_path,
-                capture=False,
-                check=True,
-            )
-            cmd_runner.run_command(
-                cmd=["tar", "--strip-components=1", "-zxf", "mongodb.tgz"],
-                cwd=mongo_repo_path,
-                capture=False,
-                check=True,
+    info = toolchain.toolchain_info(genny_repo_root=genny_repo_root, workspace_root=workspace_root)
+    if mongodb_archive_url is None:
+        if info.is_darwin:
+            artifact_key = "osx"
+        elif info.linux_distro in CANNED_ARTIFACTS:
+            artifact_key = info.linux_distro
+        else:
+            raise Exception(
+                f"No pre-built artifacts for distro {info.linux_distro}. You can either:"
+                f"1. Modify the CANNED_ARTIFACTS dict in the genny python to include an artifact from a waterfall build."
+                f"2. Pass in the --mongodb-archive-url parameter to force a canned artifact."
             )
-            mongod = from_tarball
+        mongodb_archive_url = CANNED_ARTIFACTS[artifact_key]
+
+    SLOG.info("Fetching and installing mongod.", fetching=mongodb_archive_url)
+    cmd_runner.run_command(
+        cmd=["curl", "-LSs", mongodb_archive_url, "-o", "mongodb.tgz"],
+        cwd=mongo_repo_path,
+        capture=False,
+        check=True,
+    )
+    cmd_runner.run_command(
+        cmd=["tar", "--strip-components=1", "-zxf", "mongodb.tgz"],
+        cwd=mongo_repo_path,
+        capture=False,
+        check=True,
+    )
+    mongod = from_tarball
```

Comment:
Is this downloading and extracting unconditionally? I think we want to check if it's already there, and reuse it if so (making sure it's the right version, so that python pinned version updates will overwrite the one there).

Reply:
How do I know what is the right version without extracting the artificat?

Reply:
I think we would check if there is already a downloaded and extracted artifact in the build dir. If so, we can check the version on it and use it. If there is not a binary already in the build dir, we download and extract. (Not sure if the intermediate tarball would be kept, probably not needed).

So ideally the first time I run genny it'll download and extract, then the second time it'll use the binary already there.